### PR TITLE
fix(#6338): LimitLines persist when series is moved to another Y Axis

### LINKED ISSUE #6338: Limit lines disappear when changing the y axis for series (navigating away and back solves the issue)
#### Summary
If you view an overlay plot with limit lines enabled, then change the y axis the series belongs to, the limit lines disappear.

#### Steps to Reproduce
1. Create an overlay plot with a telemetry endpoint that has limits defined.
2. Edit the plot to enable limit lines, ensure that the lines show up and save it.
3. Edit the overlay plot and move the telemetry endpoint to a different y axis (elements pool)
4. Save the change - observe that the limit lines have disappeared
5. WORKAROUND - navigate away and back to see the limit lines.

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by ozyx**
Closes #6338 

### Describe your changes:

Fixes an issue where LimitLines were being cleared when a series is moved to another Y Axis

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 1]
```diff
@@ -60,6 +60,75 @@ test.describe('Overlay Plot', () => {
...
+
+        const swgA = await createDomainObjectWithDefaults(page, {
+            type: "Sine Wave Generator",
+            parent: overlayPlot.uuid
+        });
+
+        await page.goto(overlayPlot.url);
+
+        // Assert that no limit lines are shown by default
+        await page.waitForSelector('.js-limit-area', { state: 'attached' });
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:81]: I'm wondering if there's a way we could update our best practices to ensure that folks know when to use css classes (like this) versus when to use roles/text/ids

Comment (ozyx) [reply e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:81]: Yes-- actually I'll get to doing that now. There's a lot we need to update in those example tests / best practices.

**Review by shefalijoshi — APPROVED** [review]

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
I only got 15 minutes into the review. I have one change request to follow

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 2]
```diff
@@ -60,6 +60,75 @@ test.describe('Overlay Plot', () => {
...
+        const limitLineCount = await page.locator('.c-plot-limit-line').count();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:82]: Same applies here. There is a scenario which would prefer a css locator which may not be particularly descriptive

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 3]
```diff
@@ -198,3 +267,16 @@ async function getCanvasPixelsWithData(page) {
...
     return getTelemValuePromise;
 }
+
+/**
+ *
+ * @param {import('@playwright/test').Page} page
+ */
+async function assertLimitLinesExistAndAreVisible(page) {
+    await page.waitForSelector('.js-limit-area', { state: 'attached' });
+    const limitLineCount = await page.locator('.c-plot-limit-line').count();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:277]: so you're declaring this count within the function and within the test with two separate assertions: 0 and 10.  You also use this locator on line 280.

I would consider maybe turning it into a 1 liner on all instances.

//Line limits should be 10 by default
expect(await page.locator('.c-plot-limit-line').count()).toBe(10)

Comment (ozyx) [reply e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:277]: I agree, but I actually need the variable here for the loop so I can verify that the limit lines are actually visible on the page-- so I'll probably need to keep it in this case.

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**